lib/systems/platforms.nix

```diff
@@ -1,3 +1,10 @@
+# Note: lib/systems/default.nix takes care of producing valid,
+# fully-formed "platform" values (e.g. hostPlatform, buildPlatform,
+# targetPlatform, etc) containing at least the minimal set of attrs
+# required (see types.parsedPlatform in lib/systems/parse.nix).  This
+# file takes an already-valid platform and further elaborates it with
+# optional fields such as linux-kernel, gcc, etc.
```

Comment:
Only `linux-kernel`, `gcc`, and `rustc` currently. It is unclear if we should even do those together. This file probably should be split up so we have separate auto-detectors for each of those.

Reply:
> Only `linux-kernel`, `gcc`, and `rustc` currently.

Updated the comment to reflect that.

> This file probably should be split up so we have separate auto-detectors for each of those.

I can submit a PR for that if you like.

Not having this stuff in a file named `platforms.nix` would've made it easier for me to understand what was going on.  Right now it is a bit confusing that host*Platform* is determined by code in lib/*systems*, rather than *platforms*.nix.  Renaming lib/systems/ is probably too big of a change (tons of merge conflicts) but at least not having something called `platforms.nix` would prevent people from looking there expecting to find the code that builds the attrsets that eventually get assigned to {build,host,target,cross}Platform.